src/data-management/storage-service.ts

```diff
@@ -0,0 +1,82 @@
+import { Note } from './types';
+
+export class StorageService {
+    private readonly STORAGE_KEY_PREFIX = 'note_';
+
+    private getKey(id: string): string {
+        return this.STORAGE_KEY_PREFIX + id;
+    }
+
+    async getNotes(
+        { options }: { options: { includeDeleted: boolean } } = {
+            options: { includeDeleted: false },
+        }
+    ): Promise<Note[]> {
+        return new Promise((resolve) => {
+            const notes: Note[] = [];
+            for (let i = 0; i < localStorage.length; i++) {
```

Comment:
Why do we store each note as a separate local storage record? I think it would be easier and less messy to store a single `notes` array.

Reply:
Actually, not a big deal for our goal. Disregard.

Reply:
- to add fast and get one fast
 - to not get all and then find or filter for a specific note

Reply:
@Letushev I discussed it with Yurii, lets try to keep entire notes in one record. 
For duckfooding it's okay.